internal/plugin/connectors/tcp/mssql/connection_details_test.go

```diff
@@ -108,3 +102,56 @@ func TestNewConnectionDetails(t *testing.T) {
 		})
 	}
 }
+
+func TestConnectionDetails_ParseSSLOptions(t *testing.T) {
+	type args struct {
+		credentials map[string][]byte
+	}
+	tests := []struct {
+		description string
+		args        args
+		expected    *ConnectionDetails
+	}{
+		{
+			description: "no values",
+			args: args{
+				credentials: map[string][]byte{},
+			},
+			expected: &ConnectionDetails{
+				Port:       defaultMSSQLPort,
+				SSLMode:    defaultSSLMode,
+				SSLOptions: map[string]string{},
+			},
+		},
+		{
+			description: "standard values found",
+			args: args{
+				credentials: map[string][]byte{
+					"sslrootcert": []byte("foo"),
+					"sslkey":      []byte("bar"),
+					"sslcert":     []byte("foobar"),
+				},
+			},
+			expected: &ConnectionDetails{
+				Port:    defaultMSSQLPort,
+				SSLMode: defaultSSLMode,
+				SSLOptions: map[string]string{
+					"sslrootcert": "foo",
+					"sslkey":      "bar",
+					"sslcert":     "foobar",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			actualConnDetails := NewConnectionDetails(tt.args.credentials)
+			actualConnDetails.parseSSLOptions(tt.args.credentials)
+
+			assert.Equal(t, tt.expected, actualConnDetails)
+
+			assert.Empty(t, tt.args.credentials)
```

Comment:
It's worth adding a comment above this to inform that this is validating zeroization.